DaoBuilder/daoUtils.ts

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { SafeTransaction } from "./types";
+import { buildContractCall, getRandomBytes } from "./utils";
+import { Contract } from "ethers";
+import { ethers } from "hardhat";
+import {
+  GnosisSafeProxyFactory__factory as GnosisSafeFactory,
+  Azorius as IAzorius,
+  ModuleProxyFactory as IModuleProxyFractory,
+  LinearERC20Voting as ILinearERC20Voting,
+  KeyValuePairs as IKeyValuePairs,
+  FractalRegistry as IFractalRegistry,
+  GnosisSafe,
+  GnosisSafeProxyFactory,
+} from "@fractal-framework/fractal-contracts";
+import { getCreate2Address, solidityKeccak256 } from "ethers/lib/utils";
+import {
+  getMultiSendCallOnlyDeployment,
+  getProxyFactoryDeployment,
+  getSafeSingletonDeployment,
+} from "@safe-global/safe-deployments";
+const { AddressZero, HashZero } = ethers.constants;
+
+export const CHAIN_ID = 5;
+export const SAFE_VERSION = "1.3.0";
+
+/* eslint-disable node/no-unsupported-features/es-syntax */
+async function getFractalContractAddressByChainId(_chainId: number) {
+  const networkPath = _chainId === 1 ? "mainnet" : "goerli";
+
+  const Azorius = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/Azorius.json`
+  );
+  const FractalRegistry = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/FractalRegistry.json`
+  );
+  const KeyValuePairs = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/KeyValuePairs.json`
+  );
+  const LinearERC20Voting = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/LinearERC20Voting.json`
+  );
+  const ModuleProxyFactory = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/ModuleProxyFactory.json`
+  );
+  return {
+    Azorius,
+    FractalRegistry,
+    KeyValuePairs,
+    LinearERC20Voting,
+    ModuleProxyFactory,
+  };
+}
```

Comment:
So this dynamically imports the deployment json of the current chainId. set at the top here. This will make it easier once we go to mainnet we just switch that const.